naff/models/discord/role.py

```diff
@@ -220,3 +220,32 @@ async def move(self, position: int, reason: str | Missing = MISSING) -> "Role":
             self._guild_id, [{"id": self.id, "position": position}], reason
         )
         return self
+
+    # START: fail if not all permissions are manageable
+    async def clone(self) -> "Role":
+        return await self.guild.create_role(**self.to_dict())
+
+    # END: fail if not all permissions are manageable
+
+    # START: only set manageable permissions
+    async def clone(self) -> "ClonedRole":
+        cloneable_permissions = Permissions(self.permissions.value & self.guild.me.guild_permissions.value)
+        missing_permissions = Permissions(self.permissions.value & ~self.guild.me.guild_permissions.value)
+
+        role = await self.guild.create_role(
+            name=self.name,
+            permissions=cloneable_permissions,
+            color=self.color,
+            hoist=self.hoist,
+            mentionable=self.mentionable,
+            icon=self.icon,
+        )
+        return ClonedRole.from_dict(
+            data=role.to_dict() | {"missing_permissions": missing_permissions}, client=self._client
+        )
+
+
+@attrs.define(eq=False, order=False, hash=False, kw_only=True)
+class ClonedRole(Role):
```

Comment:
Why not just wait for the role-create event and return an actual role object?

Reply:
yeah, I missed the part with the `role.id`, will fix it tomorrow (or soon:tm:)

but I'll stick to `ClonedRole` to easily get not-cloned permissions